project-koku/koku
# Cost2270 RBAC changes don't invalidate the cache

Fixes [COST-2270](https://issues.redhat.com/browse/COST-2270)

Changes proposed in this PR:
Enables us to change the rate at which the RBAC cache is invalidated.
The Default rate is at 300s or 5 min. 


Testing Instructions:
1. Set your RBAC_CACHE_TIMEOUT to a short time, in this example 5 
2. Enter the Django Shell : REDIS_HOST=localhost make shell
3.Run These set of commands 
```
- from django.core.cache import caches
- cache = caches['rbac']
- cache.set('my_key', 'hello, world!')
- cache.get('my_key') : expected result 'hello, world!'
```
after your set time has passed, rerun the command 
`- cache.get('my_key') : expected result should be no output`
4. Make sure your settings.CACHE has been updated 
```
- from django.conf import settings
- settings.CACHES
```
Example output 

> `{'default': {'BACKEND': 'django.core.cache.backends.dummy.DummyCache'}, 'rbac': {'BACKEND': 'django_redis.cache.RedisCache', 'LOCATION': 'redis://localhost:6379/1', 'OPTIONS': {'CLIENT_CLASS': 'django_redis.client.DefaultClient', 'IGNORE_EXCEPTIONS': True, 'MAX_ENTRIES': 1000}}, 'worker': {'BACKEND': 'django.core.cache.backends.db.DatabaseCache', 'LOCATION': 'worker_cache_table', 'TIMEOUT': 86400}}`

---
Additional Context: Please reach out to me with questions or assistance testing the PR!
